src/io/fstream.cpp

```diff
@@ -13,11 +13,11 @@
 using namespace std;
 
 namespace fc {
-   class ofstream::impl : public fc::retainable {
```

Comment:
Why no longer retain the I/O related stuff (I/O streams, logger, log appenders, UDP socket)?

Reply:
`fc::shared_ptr` could only point to `retainable` objects. `std::shared_ptr` doesn't have that restriction, so there is no reason to use `fc::retainable` anymore. (Except for such hacks as with fc::task.)